### PR TITLE
CBPF Classifier Integration

### PR DESCRIPTION
**Pull request by and-santos**

### CODE REVIEW

File: fink_science/cbpf_classifier/processor.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+    for i, mjds in enumerate(midpointTai):
+        
+        if len(mjds) > 0:
+            
+            bands.append(np.array(
+                [filter_dict[f] for f in filterName[i]]
+            ).astype(np.int16))        
+            lc = np.concatenate(
+                [mjds[:,None], psFlux[i][:,None], psFluxErr[:,None]], axis=-1
```
Comment (JulienPeloton) [inline fink_science/cbpf_classifier/processor.py:88]: There is a `[i]` missing for `psFluxErr`. It should be:

```python
[mjds[:,None], psFlux[i][:,None], psFluxErr[i][:,None]], axis=-1
```

**Review by JulienPeloton — CHANGES_REQUESTED** [review]
Thanks for the code addition! As it stands, the code is not working. I left several comments to fix it, and I tried this code on my side successfully, except one last error:

```python
PythonException: 
  An exception was thrown from the Python worker. Please see the stack trace below.
Traceback (most recent call last):
  File "/opt/spark-3/python/lib/pyspark.zip/pyspark/worker.py", line 604, in main
    process()
  File "/opt/spark-3/python/lib/pyspark.zip/pyspark/worker.py", line 596, in process
    serializer.dump_stream(out_iter, outfile)
  File "/opt/spark-3/python/lib/pyspark.zip/pyspark/sql/pandas/serializers.py", line 273, in dump_stream
    return ArrowStreamSerializer.dump_stream(self, init_stream_yield_batches(), stream)
  File "/opt/spark-3/python/lib/pyspark.zip/pyspark/sql/pandas/serializers.py", line 81, in dump_stream
    for batch in iterator:
  File "/opt/spark-3/python/lib/pyspark.zip/pyspark/sql/pandas/serializers.py", line 266, in init_stream_yield_batches
    for series in iterator:
  File "/opt/spark-3/python/lib/pyspark.zip/pyspark/worker.py", line 450, in mapper
    result = tuple(f(*[a[o] for o in arg_offsets]) for (arg_offsets, f) in udfs)
  File "/opt/spark-3/python/lib/pyspark.zip/pyspark/worker.py", line 450, in <genexpr>
    result = tuple(f(*[a[o] for o in arg_offsets]) for (arg_offsets, f) in udfs)
  File "/opt/spark-3/python/lib/pyspark.zip/pyspark/worker.py", line 105, in <lambda>
    verify_result_type(f(*a)), len(a[0])), arrow_return_type)
  File "/opt/spark-3/python/lib/pyspark.zip/pyspark/util.py", line 73, in wrapper
    return f(*args, **kwargs)
  File "<ipython-input-2-84f9cf2cacbf>", line 114, in predict_nn
  File "/opt/anaconda/lib/python3.7/site-packages/keras/utils/traceback_utils.py", line 67, in error_handler
    raise e.with_traceback(filtered_tb) from None
  File "/opt/anaconda/lib/python3.7/site-packages/keras/utils/generic_utils.py", line 563, in class_and_config_for_serialized_keras_object
    f'Unknown {printable_module_name}: {class_name}. Please ensure this '
ValueError: Unknown optimizer: Addons>RectifiedAdam. Please ensure this object is passed to the `custom_objects` argument. See https://www.tensorflow.org/guide/keras/save_and_serialize#registering_the_custom_object for details.
```

As if `RectifiedAdam` was not known. Please can you check this? 

Also, `fink-science` should be as lightweight as possible to be distributed easily. So you should remove from this PR all big and unnecessary files:
- the folder `cbpf_classifier/data` (this data should not be on the repo)
- the folder containing submodules `src/`

**Comment by and-santos** [comment]
Hi Julien, I made all the suggested changes and exlude de src and data folders.

The problem with RectifiedAdam is that custom optimizers should be explicited when loading a model. This is solved calling custom_objects inside load_model function call with the custom optimizer.

I made a new commit fixing the RectifiedAdam error together with the other suggestions. Can you try again to see if works?

File: fink_science/cbpf_classifier/processor.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+                [mjds[:, None], psFlux.values[i][:, None],
+                    psFluxErr.values[i][:, None]], axis=-1
+            )
+
+            if not np.isnan(mwebv.values[i]):
+
+                lcs.append(normalize_lc(lc).astype(np.float32))
+
+                meta.append([
+                    mwebv.valaues[i], z_final.vaalues[i],
```
Comment (JulienPeloton) [inline fink_science/cbpf_classifier/processor.py:97]: There is a typo in the names:

```diff
-  mwebv.valaues[i], z_final.vaalues[i],
+  mwebv.values[i], z_final.values[i],
```

**Review by JulienPeloton — COMMENTED** [review]
Thanks @and-santos ! Except one typo left (see comment below), it works like a charm :-) I could run it on test alerts, and get predictions. I will generate predictions for many alerts, and share with you for inspection.

**Comment by JulienPeloton** [comment]
@and-santos -- would it be possible to correct the failures due to PEP8?
https://github.com/astrolabsoftware/fink-science/actions/runs/3052725523/jobs/4922463872
Thanks!

**Comment by and-santos** [comment]
@JulienPeloton I made the corrections following PEP8. Everything should work now :)

**Comment by JulienPeloton** [comment]
Hi @and-santos -- Merging these developments (tests are failing for a reason unrelated to the CBPF addition). Do not hesitate to open a new PR if you want to push more.

**Pull request merged**